homeassistant/requirements.py

```diff
@@ -146,22 +159,52 @@ async def async_process_requirements(
     pip_lock = hass.data.get(DATA_PIP_LOCK)
     if pip_lock is None:
         pip_lock = hass.data[DATA_PIP_LOCK] = asyncio.Lock()
+    install_failure_history = hass.data.get(DATA_INSTALL_FAILURE_HISTORY)
+    if install_failure_history is None:
+        install_failure_history = hass.data[DATA_INSTALL_FAILURE_HISTORY] = {}
 
     kwargs = pip_kwargs(hass.config.config_dir)
 
     async with pip_lock:
         for req in requirements:
-            if pkg_util.is_installed(req):
-                continue
+            await _async_process_requirements(
+                hass, name, req, install_failure_history, kwargs
+            )
 
-            def _install(req: str, kwargs: dict[str, Any]) -> bool:
-                """Install requirement."""
-                return pkg_util.install_package(req, **kwargs)
 
-            ret = await hass.async_add_executor_job(_install, req, kwargs)
+async def _async_process_requirements(
+    hass: HomeAssistant,
+    name: str,
+    req: str,
+    install_failure_history: dict[str, datetime],
+    kwargs: Any,
+) -> None:
+    """Install a requirement and save failures."""
+    if pkg_util.is_installed(req):
+        return
 
-            if not ret:
-                raise RequirementsNotFound(name, [req])
+    def _install(req: str, kwargs: dict[str, Any]) -> bool:
+        """Install requirement."""
+        return pkg_util.install_package(req, **kwargs)
+
+    if req in install_failure_history:
+        retry_time = install_failure_history[req] + timedelta(hours=24)
```

Comment:
I think that we can actually do it without a timer? If it fails, mark it as failed until you reboot HA. Integration set up will already be marked as failed too.

Reply:
The line of thought was that if internet was down for a day, maybe it would be up tomorrow, and they didn't want to restart... but probably ok to just restart